pages/workspaces/components/WorkspaceHeading/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { useUpdateWorkspaceMutation } from '@/graphql/workspace/UpdateWorkspace.generated';
+import { Heading, Spinner, Input } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+
+const WorkspaceHeading: React.FC<{
+  workspaceName?: string;
+  workspaceId: string;
+  loading: boolean;
+}> = ({ workspaceName, workspaceId, loading }) => {
+  const [isEditingName, setIsEditingName] = useState(false);
+  const [newName, setNewName] = useState(workspaceName ? workspaceName : '');
+  const [afterUpdateName, setAfterUpdateName] = useState('');
+  const [mutate] = useUpdateWorkspaceMutation();
+
+  const makeUpdate = async () => {
+    await mutate({ variables: { id: workspaceId, name: newName } });
+  };
+
+  // this is so the input element starts with the existing workspace name
+  // (which isn't immediately available)
+  useEffect(() => {
+    if (!workspaceName) {
+      return;
+    }
+
+    setNewName(workspaceName);
+  }, [workspaceName]);
```

Comment:
Can't this be:

```suggestion
  useEffect(() => {
    setNewName(workspaceName || '');
  }, [workspaceName]);
```

Reply:
I'm a fan of being explicit in this type of situation - perhaps the implicit setting of `''` which isn't a change which may not cause a re-render is more common than I know of in react land but my first thought is "eh now I have to figure out if setting it to empty string is going to do anything or not"